ballerina/modules/wssec/ws_security_methods.bal

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/crypto;
+import ballerina/regex;
+
+xmlns "http://schemas.xmlsoap.org/soap/envelope/" as soap;
+xmlns "http://www.w3.org/2000/09/xmldsig#" as ds;
+
+function addSecurityHeader(Document document) returns WSSecurityHeader|Error {
+    WSSecurityHeader wsSecHeader = check new (document);
+    Error? insertHeader = wsSecHeader.insertSecHeader();
+    if insertHeader is () {
+        return wsSecHeader;
+    }
+    return insertHeader;
```

Comment:
```suggestion
    return insertHeader ?: wsSecHeader;
```
We could do this too

Reply:
Fixed in [1de6750](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/1de6750f3a32d73e90b0ade50476774f2541fccf)